### ISSUE TITLE
Overlap with empty range incorrectly returns True

### ISSUE THREAD

**Issue by gson1703** [runfalk/spans#7]
The overlap() method is documented as "Returns True if both ranges share any points."
The following example incorrectly prints True even though the ranges clearly do not share any points.

from spans import intrange
print intrange(20, 30).overlap(intrange(10, 10))

**Comment by runfalk** [comment]
This is indeed a bug. PostgreSQL does:

```
 # SELECT 'empty'::int4range && 'empty'::int4range;
 ?column? 
----------
 f
(1 row)
```

or

```
# SELECT int4range(1, 10, '[)') && 'empty'::int4range;
 ?column? 
----------
 f
(1 row)
```

**Comment by runfalk** [comment]
I've made a fix for this. Please test it and report back if it fixes your issue. The only changes file is `types.py`.

**Comment by gson1703** [comment]
The fix looks good to me.  Thanks for the quick response!

**Comment by runfalk** [comment]
I'll release this fix as part of the 1.0 release. It'll probably be another week or two before that in case any other issues arise.

**Comment by runfalk** [comment]
1.0.0 is now released.
